lib/racc/parser.rb

```diff
@@ -246,24 +245,24 @@ def _racc_do_parse_rb(arg, in_debug)
       _racc_init_sysvars
       tok = act = i = nil
 
-      catch(:racc_end_parse) {
-        while true
+      catch(:racc_end_parse) do
+        loop do
           if i = action_pointer[@racc_state[-1]]
             if @racc_read_next
-              if @racc_t != 0   # not EOF
-                tok, @racc_val = next_token()
-                unless tok      # EOF
-                  @racc_t = 0
-                else
-                  @racc_t = (token_table[tok] or 1)   # error token
+              if @racc_t != 0 # not EOF
+                tok, @racc_val = next_token
+                @racc_t = unless tok # EOF
+                            0
+                          else
+                            (token_table[tok] || 1) # error token
                 end
```

Comment:
This `end` should be aligned with the `unless` and `else`. You could file a bug against RuboCop for that.